### PR TITLE
e2e: Allow testing custom cni and dp image

### PR DESCRIPTION
**Pull request by zeeke**
Let `run-e2e-conformance-virtual-ocp.sh` and
`run-e2e-conformance-virtual-cluster.sh` deploy clusters with specific SRIOV CNI and Device Plugin images. This will enable other projects to have full end-to-end test CI with the operator's suite.

Ref:
- https://github.com/k8snetworkplumbingwg/sriov-cni/pull/282

cc @adrianchiris @e0ne

### CODE REVIEW

File: hack/run-e2e-conformance-virtual-ocp.sh [hunk 1]
```diff
@@ -189,7 +189,7 @@ echo "## build daemon image"
 podman build -t "${SRIOV_NETWORK_CONFIG_DAEMON_IMAGE}" -f "${root}/Dockerfile.sriov-network-config-daemon" "${root}"
 
 echo "## build webhook image"
-podman build -t "${SRIOV_NETWORK_WEBHOOK_IMAGE}" -f "${root}/Dockerfile.webhook" "${root}"
+podman `build` -t "${SRIOV_NETWORK_WEBHOOK_IMAGE}" -f "${root}/Dockerfile.webhook" "${root}"
```
Comment (bn222) [inline hack/run-e2e-conformance-virtual-ocp.sh:192]: why quotes around build?

Comment (zeeke) [reply hack/run-e2e-conformance-virtual-ocp.sh:192]: That went by mistake. Thanks for catching it

File: hack/run-e2e-conformance-virtual-ocp.sh [hunk 2]
```diff
@@ -204,6 +204,20 @@ podman push --tls-verify=false "${SRIOV_NETWORK_OPERATOR_IMAGE}"
...
+  podman tag ${LOCAL_SRIOV_CNI_IMAGE} ${SRIOV_CNI_IMAGE}
+  podman push ${SRIOV_CNI_IMAGE}
+fi
+
+if [[ -v LOCAL_SRIOV_DEVICE_PLUGIN_IMAGE ]]; then
+  export SRIOV_DEVICE_PLUGIN_IMAGE="$registry/$NAMESPACE/sriov-network-device-plugin:latest"
+  echo "## Pushing ${LOCAL_SRIOV_DEVICE_PLUGIN_IMAGE} to cluster iamge ${SRIOV_SRIOV_DEVICE_PLUGIN_IMAGE}"
+  podman tag ${LOCAL_SRIOV_DEVICE_PLUGIN_IMAGE} ${SRIOV_SRIOV_DEVICE_PLUGIN_IMAGE}
+  podman push ${SRIOV_DEVICE_PLUGIN_IMAGE}
+fi
```
[the remark this answers is not in the document]
Comment (zeeke) [reply hack/run-e2e-conformance-virtual-ocp.sh:216]: done!

**Comment by zeeke** [comment]
/hold
need to be tested with
- https://github.com/k8snetworkplumbingwg/sriov-cni/pull/282

File: hack/run-e2e-conformance-virtual-cluster.sh [hunk 3]
```diff
@@ -254,6 +258,16 @@ podman push --tls-verify=false "${SRIOV_NETWORK_OPERATOR_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_CONFIG_DAEMON_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_WEBHOOK_IMAGE}"
 
+
```
Comment (SchSeba) [inline hack/run-e2e-conformance-virtual-cluster.sh:261]: can you please add the network resource injector also?

File: hack/run-e2e-conformance-virtual-ocp.sh [hunk 4]
```diff
@@ -204,6 +206,16 @@ podman push --tls-verify=false "${SRIOV_NETWORK_OPERATOR_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_CONFIG_DAEMON_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_WEBHOOK_IMAGE}"
 
+
```
Comment (SchSeba) [inline hack/run-e2e-conformance-virtual-ocp.sh:209]: can you please add the network resource injector also?

Comment (SchSeba) [inline hack/run-e2e-conformance-virtual-ocp.sh:209]: you also need to add in line 240 the export

Comment (zeeke) [reply hack/run-e2e-conformance-virtual-ocp.sh:209]: sure

Comment (zeeke) [reply hack/run-e2e-conformance-virtual-ocp.sh:209]: right!

Comment (SchSeba) [reply hack/run-e2e-conformance-virtual-ocp.sh:209]: you didn't update here https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/518/files#diff-90e627cba9db282e04456e7e550bd478d2c12fbbbce4104f9421d640eb3a6447R247

Comment (zeeke) [reply hack/run-e2e-conformance-virtual-ocp.sh:209]: Image env variable is already exported here
https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/518/files#diff-90e627cba9db282e04456e7e550bd478d2c12fbbbce4104f9421d640eb3a6447R212 

What need to be done on L247?

**Comment by zeeke** [comment]
This is now ready for review, it worked against sriov-cni wip PR:

- https://github.com/k8snetworkplumbingwg/sriov-cni/pull/282

File: hack/run-e2e-conformance-virtual-cluster.sh [hunk 5]
```diff
@@ -386,9 +407,17 @@ if [ -z $SKIP_TEST ]; then
     export JUNIT_OUTPUT="${root}/${TEST_REPORT_PATH}/conformance-test-report"
   fi
 
+  # Disable exit on error temporarily to gather cluster information
```
Comment (SchSeba) [inline hack/run-e2e-conformance-virtual-cluster.sh:410]: should you remove this one?

File: hack/run-e2e-conformance-virtual-ocp.sh [hunk 6]
```diff
@@ -254,9 +274,17 @@ if [ -z $SKIP_TEST ]; then
     export JUNIT_OUTPUT="${root}/${TEST_REPORT_PATH}/conformance-test-report"
   fi
 
+  # Disable exit on error temporarily to gather cluster information
+  set +e
```
Comment (SchSeba) [inline hack/run-e2e-conformance-virtual-ocp.sh:278]: should you remove this one?

File: hack/run-e2e-conformance-virtual-cluster.sh (hunk 5 above)
Comment (zeeke) [reply hack/run-e2e-conformance-virtual-cluster.sh:410]: It's intentional: with `set -e`, if `run-e2e-conformance.sh` fails the script stops and no cluster info is gathered. I could use a trap, but it has been already used in L48 and trapping multiple times would create a mess in the script (see [SO](https://stackoverflow.com/questions/3338030/multiple-bash-traps-for-the-same-signal)).

Proposals:
a. I can move these changes to a different PR, as they are a little out of scope.
b. I can move the `cluster-info dump` command into `run-e2e-conformance.sh`, binding it to a trap

**Review by SchSeba — APPROVED** [review]
LGTM

thanks!

File: hack/run-e2e-conformance-virtual-cluster.sh (hunk 5 above)
Comment (SchSeba) [reply hack/run-e2e-conformance-virtual-cluster.sh:410]: sounds good we can leave it here

File: hack/run-e2e-conformance-virtual-ocp.sh [hunk 7]
```diff
@@ -218,6 +220,9 @@ podman push --tls-verify=false "${SRIOV_NETWORK_OPERATOR_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_CONFIG_DAEMON_IMAGE}"
 podman push --tls-verify=false "${SRIOV_NETWORK_WEBHOOK_IMAGE}"
 
+
+
+
```
Comment (adrianchiris) [inline hack/run-e2e-conformance-virtual-ocp.sh:225]: do we need so many newlines ?

**Review by adrianchiris — COMMENTED** [review]
minor its otherwise lgtm

File: hack/run-e2e-conformance-virtual-cluster.sh [hunk 8]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
+
 set -xeo pipefail
 
+
```
[the remark this answers is not in the document]
Comment (zeeke) [reply hack/run-e2e-conformance-virtual-cluster.sh:5]: fixing!

**Review by adrianchiris — APPROVED** [review]

**Comment by zeeke** [comment]
Two approvals gathered.
Rebasing to verify e2e tests are ok.

**Comment by zeeke** [comment]
All CI checks green. Merging

**Pull request merged**